content/en/blog/2024/trustee-deployment.md

```diff
@@ -0,0 +1,351 @@
+localityName                = Locality Name (eg, city)
+localityName_default        = Bristol
+organizationName            = Organization Name (eg, company)
+organizationName_default    = Red Hat
+organizationalUnitName      = organizationalunit
+organizationalUnitName_default = Development
+commonName                  = Common Name (e.g. server FQDN or YOUR name)
+commonName_default          = kbs-service
+commonName_max              = 64
+
+[req_ext]
+subjectAltName = @alt_names
+
+[v3_ca]
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1   = kbs-service
+EOF
+```
+
+Create secret for self-signed certificate:
+
+```bash
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout https.key -out https.crt \
+    -config kbs-service-509.conf -passin pass:\
+    -subj "/C=UK/ST=England/L=Bristol/O=Red Hat/OU=Development/CN=kbs-service"
+kubectl create secret generic kbs-https-certificate --from-file=https.crt -n kbs-operator-system
+kubectl create secret generic kbs-https-key --from-file=https.key -n kbs-operator-system
+```
+
+### Trustee ConfigMap object
+
+This command will create the ConfigMap object that provides Trustee all the needed configuration:
+
+```bash
+kubectl apply -f - << EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kbs-config
+  namespace: kbs-operator-system
+data:
+  kbs-config.json: |
+    {
+        "insecure_http" : false,
+        "private_key": "/etc/https-key/https.key",
+        "certificate": "/etc/https-cert/https.crt",
+        "sockets": ["0.0.0.0:8080"],
+        "auth_public_key": "/etc/auth-secret/publicKey",
+        "attestation_token_config": {
+          "attestation_token_type": "CoCo"
+        },
+        "repository_config": {
+          "type": "LocalFs",
+          "dir_path": "/opt/confidential-containers/kbs/repository"
+        },
+        "as_config": {
+          "work_dir": "/opt/confidential-containers/attestation-service",
+          "policy_engine": "opa",
+          "attestation_token_broker": "Simple",
+          "attestation_token_config": {
+            "duration_min": 5
+          },
+          "rvps_config": {
+            "store_type": "LocalJson",
+            "store_config": {
+              "file_path": "/opt/confidential-containers/rvps/reference-values/reference-values.json"
+            }
+          }
+        },
+        "policy_engine_config": {
+          "policy_path": "/opt/confidential-containers/opa/policy.rego"
+        }
+    }
+EOF
+```
+
+### Reference Values
+
+The reference values are an important part of the attestation process. The client collects the measurements from the running software and submits a quote with the claims to the attestation server. These measurements, in order for the attestation protocol to succeed, have to match exactly with the trusted digests that had been registered to Trustee previously. This process guarantees the cVM (confidential VM) is running the expected software stack and that it hasn’t been tampered with.
```

Comment:
in RVPS it would be not "exactly" one digest, but one of potentially multiple configured valid values. Also, the client doesn't only contain measurements from the software, but also the TEE hardware and its fw, so you could apply flexible rules ("fw of secure processor to be > v1.30"), all of those together are "evidence". e.g. In the case of an SNP TEE there would be only a single composite "launch measurement" in the evidence.

Reply:
thank you, I'll add these notes